test/test_suite.py

```diff
@@ -208,3 +222,136 @@ def test_underscore_names(self):
 @pytest.mark.skip("Max identifier length in Spanner is 128")
 class LongNameBlowoutTest(_LongNameBlowoutTest):
     pass
+
+
+class ComponentReflectionTest(_ComponentReflectionTest):
+    @classmethod
+    def define_temp_tables(cls, metadata):
+        """
+        SPANNER OVERRIDE:
+
+        In Cloud Spanner unique indexes are used instead of directly
+        creating unique constraints. Overriding the test to replace
+        constraints with indexes in testing data.
+        """
+        kw = temp_table_keyword_args(config, config.db)
+        user_tmp = Table(
+            "user_tmp",
+            metadata,
+            Column("id", sqlalchemy.INT, primary_key=True),
+            Column("name", sqlalchemy.VARCHAR(50)),
+            Column("foo", sqlalchemy.INT),
+            sqlalchemy.Index("user_tmp_uq", "name", unique=True),
+            sqlalchemy.Index("user_tmp_ix", "foo"),
+            **kw
+        )
+        if (
+            testing.requires.view_reflection.enabled
+            and testing.requires.temporary_views.enabled
+        ):
+            event.listen(
+                user_tmp,
+                "after_create",
+                DDL("create temporary view user_tmp_v as " "select * from user_tmp"),
+            )
+            event.listen(user_tmp, "before_drop", DDL("drop view user_tmp_v"))
+
+    @testing.provide_metadata
+    def _test_get_unique_constraints(self, schema=None):
```

Comment:
This test is not passing yet because of another error: one of the test columns is called `asc`, which causes syntax error. Without that column the test seems to be working fine, as unique constraints are replaced with unique indexes on the line 299, no other changes done.